src/components/CountryDetails/index.js

```diff
@@ -0,0 +1,79 @@
+import { useEffect, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { getPinned, showDisplayedCountries } from '../../store/countries/actions';
+import Flex from './../../assets/styledComponents/Flex';
+import './style.scss'
+
+const CountryDetails = ( {location} ) => {
+    const dispatch = useDispatch()
+    const {displayedCountries, pinnedCountries} = useSelector(state => state.countryReducer)
+    const [id, setId] = useState('')
+    
+    useEffect(() => {
+        const params = new URLSearchParams(location.search)
+        const id = params.get('id')
+        if (!JSON.parse(sessionStorage.getItem('display'))) {
+            sessionStorage.setItem('display', '[]')
+        } else if (!JSON.parse(sessionStorage.getItem('pinned'))) {
+            sessionStorage.setItem('pinned', '[]')
+        }
+        dispatch(showDisplayedCountries(JSON.parse(sessionStorage.getItem('display'))))
+        dispatch(getPinned(JSON.parse(sessionStorage.getItem('pinned'))))
+        setId(id)
+    }, [])
+
+
+    const displayTable = (id) => {
+        const filteredArr = displayedCountries.filter(item => item.id === id)
+        if (filteredArr.length) {
+            return filteredArr.map(item => item)
```

Comment:
there is no point  to go again through the list  again ,  you could just return the filtered list, same thing to the bottom